tree/dataframe/inc/ROOT/RDF/GraphNode.hxx

```diff
@@ -63,12 +64,20 @@ private:
 public:
    ////////////////////////////////////////////////////////////////////////////
    /// \brief Creates a node with a name and a counter
-   GraphNode(const std::string_view &name) : fName(name) { fCounter = GetStaticGlobalCounter()++; }
+   GraphNode(const std::string_view &name) : fName(name)
+   {
+      R__LOCKGUARD(gROOTMutex);
+      fCounter = GetStaticGlobalCounter()++;
+   }
 
    ////////////////////////////////////////////////////////////////////////////
    /// \brief Resets the counter.
    /// This is not strictly needed but guarantees that two consecutive request to the graph return the same result.
-   static void ClearCounter() { GraphNode::GetStaticGlobalCounter() = 1; }
+   static void ClearCounter()
+   {
+      R__LOCKGUARD(gROOTMutex);
+      GraphNode::GetStaticGlobalCounter() = 1;
```

Comment:
if two threads interleave calls to `GraphNode(const std::string_view &name)` and `ClearCounter`, one of the threads might end up with multiple `GraphNode`s with the same counter. I don't how how much of a problem that is, I'd need to check.

Reply:
also maybe we can avoid having to put two locks by just making the static global counter an atomic